index.bs

```diff
@@ -387,233 +507,172 @@ Each \[[Stage]] value indicates the currently pending operation. If the \[[Resul
-  </tr>
-</table>
-
-<h3 id="registry-constructor">The Registry Constructor</h3>
-
-The Registry constructor is the %Registry% intrinsic object. When called as a constructor it creates and initializes a new %Registry% object. %Registry% is not intended to be called as a function and will throw an exception when called in that manner.
-
-<h4 id="new-registry" aoid="Registry">Registry(loader)</h4>
+<h4 id="new-registry-entry" aoid="RegistryEntry">RegistryEntry(loader)</h4>
 
 When Registry is called with argument <i>loader</i>, the following steps are taken:
 
 <emu-alg>
 1. If NewTarget is *undefined*, then throw a *TypeError* exception.
 1. If Type(_loader_) is not Object, throw a *TypeError* exception.
-1. Let _O_ be OrdinaryCreateFromConstructor(NewTarget, "%RegistryPrototype%", «[[RegistryData]]» ).
+1. Let _O_ be OrdinaryCreateFromConstructor(NewTarget, "%RegistryPrototype%", «[[Loader]], [[Pipeline]], [[Module]], [[Metadata]], [[Dependencies]], [[Error]]» ).
 1. ReturnIfAbrupt(_O_).
-1. Set _O_’s [[RegistryData]] internal slot to a new empty List.
+1. Let _pipeline_ be a new List.
+1. Add new stage entry record { [[Stage]]: "fetch", [[Result]]: *undefined* } as a new element of the list _pipeline_.
+1. Add new stage entry record { [[Stage]]: "translate", [[Result]]: *undefined* } as a new element of the list _pipeline_.
+1. Add new stage entry record { [[Stage]]: "instantiate", [[Result]]: *undefined* } as a new element of the list _pipeline_.
+1. Add new stage entry record { [[Stage]]: "satisfy", [[Result]]: *undefined* } as a new element of the list _pipeline_.
+1. Add new stage entry record { [[Stage]]: "link", [[Result]]: *undefined* } as a new element of the list _pipeline_.
+1. Add new stage entry record { [[Stage]]: "ready", [[Result]]: *undefined* } as a new element of the list _pipeline_.
 1. Set _O_’s [[Loader]] internal slot to _loader_.
+1. Set _O_’s [[Pipeline]] internal slot to _pipeline_.
+1. Set _O_’s [[Module]] internal slot to *undefined*.
+1. Set _O_’s [[Metadata]] internal slot to *undefined*.
+1. Set _O_’s [[Dependencies]] internal slot to *undefined*.
+1. Set _O_’s [[Error]] internal slot to *nothing*.
 1. Return _O_.
 </emu-alg>
 
-<h3 id="properties-of-the-registry-constructor">Properties of the Registry Constructor</h3>
+<emu-note>
+  A registry entry is associated to a loader instance at the time of its creation, the [[Loader]] backpointer reflects that.
+</emu-note>
+
+<emu-note>
+  A registry entry can be set into multiple Register instances.
+</emu-note>
+
+<h3 id="properties-of-the-registry-entry-constructor">Properties of the Registry Entry Constructor</h3>
 
 The value of the \[[Prototype]] internal slot of the Registry constructor is the intrinsic object %FunctionPrototype%.
 
 Besides the internal slots and the length property (whose value is 0), the Registry constructor has the following properties:
```

Comment:
Shouldn't the value of `length` be 1 since it expects a single argument (the loader)?